File: 1-로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    cnt1 = 0
    cnt2 = 0
    for i in lottos:
        if i in win_nums:
            cnt1 += 1
        elif i == 0:
            cnt2 += 1
    if cnt1 == 0:
        cnt1 += 1
    up = 7 - (cnt1 + cnt2)
    down = 7 - cnt1
    answer = [up, down]
    if cnt2 == 6:
        answer = [1, 6]
        
    return answer

File: programers/1-로또의최고순위와최저순위.js
function solution(lottos, win_nums) {
    var answer = [];
    
    const good = lottos.filter(v => win_nums.includes(v)).length
    const zero = lottos.filter(v => v === 0).length
    
    var luck = good + zero
    
    var min = 7-good >= 6 ? 6 : 7-good;
    var max = good+zero === 0 ? 6 : 7-(good+zero)
    
    answer.push(max, min)
    
    return answer;
}


// function solution(lottos, win_nums) {
//     var answer = [];
//     var min = 0;
//     var max = 0;
    
//     for(var i = 0; i < 6; i++){
//         if(win_nums.includes(lottos[i])){
//             min += 1;
//         } else if(lottos[i] === 0){
//             max += 1;
//         }
//     }
//     max = max + min;
    
//     if(min === 0 && max === 6){
//         answer.push(1, 6);
//     } else if(min === 0 && max === 0){
//         answer.push(6,6)
//     }else if(min === 6){
//         answer.push(1, 1)
//     } else{
//         answer.push(7-max, 7-min)
//     }
    
//     return answer;
// }